pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -250,20 +252,22 @@ func (s *Service) Handle(ctx context.Context, stream p2p.Stream, remoteMultiaddr
 		return nil, errors.New("no observed underlay sent")
 	}
 
-	// handle only the first underlay, for now
-	observedUnderlay := observedUnderlays[0]
-
-	advertisableUnderlay, err := s.advertisableAddresser.Resolve(observedUnderlay)
-	if err != nil {
-		return nil, err
+	advertisableUnderlays := make([]ma.Multiaddr, len(observedUnderlays))
+	for i, observedUnderlay := range observedUnderlays {
+		advertisableUnderlay, err := s.advertisableAddresser.Resolve(observedUnderlay)
+		if err != nil {
+			return nil, err
+		}
+		advertisableUnderlays[i] = advertisableUnderlay
 	}
 
-	bzzAddress, err := bzz.NewAddress(s.signer, advertisableUnderlay, s.overlay, s.networkID, s.nonce)
+	bzzAddress, err := bzz.NewAddress(s.signer, advertisableUnderlays, s.overlay, s.networkID, s.nonce)
 	if err != nil {
 		return nil, err
 	}
 
-	advertisableUnderlayBytes, err := bzzAddress.Underlay.MarshalBinary()
+	// TODO support multiple underlays
+	advertisableUnderlayBytes, err := bzzAddress.Underlay[0].MarshalBinary()
```

Comment:
This should contain all underlay addresses, not just the first one. It should use the SerializeUnderlays function to serialzie them all and in the handler it should use DeserializeUnderlays.